client/src/modules/cash/modals/modal-cancel-cash.ctrl.js

```diff
@@ -2,36 +2,58 @@ angular.module('bhima.controllers')
   .controller('ModalCancelCashController', ModalCancelCashController);
 
 ModalCancelCashController.$inject = [
-  '$uibModalInstance', 'CashService', 'data', 'VoucherService', 'NotifyService'
+  '$filter', '$state', '$uibModalInstance',
+  'CashService', 'data', 'VoucherService', 'NotifyService',
 ];
 
-function ModalCancelCashController(Instance, Cash, data, Vouchers, Notify) {
-  var vm = this;
+function ModalCancelCashController($filter, $state, Instance, Cash, data, Vouchers, Notify) {
+  const vm = this;
 
   vm.cancelCash = {};
   vm.submit = submit;
-  vm.cancel = function () { Instance.close(false); };
+  vm.goToPatientLink = goToPatientLink;
+  const $currency = $filter('currency');
+
+  vm.cancel = () => Instance.close(false);
 
   vm.cancelCash.uuid = data.invoice.uuid;
   vm.patientInvoice = data.invoice;
 
   Cash.read(data.invoice.uuid)
-    .then(function (response) {
+    .then((response) => {
       vm.cashData = response;
+      vm.alertI18nValues = {
+        invoiceReference : vm.cashData.reference,
+        patientName : vm.cashData.debtorName,
+        patientReference : vm.cashData.debtorReference,
+        cost : $currency(vm.cashData.amount, vm.cashData.currency_id),
+      };
     })
     .catch(Notify.handleError);
 
   function submit(form) {
-     // stop submission if the form is invalid
+    // stop submission if the form is invalid
     if (form.$invalid) { return; }
 
-    return Vouchers.reverse(vm.cancelCash)
-      .then(function () {
+    Vouchers.reverse(vm.cancelCash)
```

Comment:
By removing the `return` of the promise, this breaks the loading indicator on the submit button.  It would be better to put this back and add a "0" to the other `return`.